test/nstemplatetier/nstemplatetier.go

```diff
@@ -187,6 +196,19 @@ func Tier(t *testing.T, name string, spec toolchainv1alpha1.NSTemplateTierSpec,
 // TierOption an option to configure the NStemplateTier
 type TierOption func(*toolchainv1alpha1.NSTemplateTier)
 
+func WithStatusRevisions(tierSpec toolchainv1alpha1.NSTemplateTierSpec) TierOption {
+	return func(tier *toolchainv1alpha1.NSTemplateTier) {
+		tier.Status.Revisions = make(map[string]string)
+		for _, ns := range tierSpec.Namespaces {
+			tier.Status.Revisions[ns.TemplateRef] = ns.TemplateRef + "-cr"
+		}
+		tier.Status.Revisions[tierSpec.ClusterResources.TemplateRef] = tierSpec.ClusterResources.TemplateRef + "-cr"
+		for _, sp := range tierSpec.SpaceRoles {
+			tier.Status.Revisions[sp.TemplateRef] = sp.TemplateRef + "-cr"
+		}
+	}
+}
```

Comment:
As I mentioned in the comment and also during the call, you can generate the revisions dynamically, you don't need to provide any parameter:
```suggestion
func WithStatusRevisions() TierOption {
	return func(tier *toolchainv1alpha1.NSTemplateTier) {
		tier.Status.Revisions = make(map[string]string)
		for _, ns := range tier.Spec.Namespaces {
			tier.Status.Revisions[ns.TemplateRef] = ns.TemplateRef + "-cr"
		}
		tier.Status.Revisions[tier.Spec.ClusterResources.TemplateRef] = tier.Spec.ClusterResources.TemplateRef + "-cr"
		for _, sp := range tier.Spec.SpaceRoles {
			tier.Status.Revisions[sp.TemplateRef] = sp.TemplateRef + "-cr"
		}
	}
}
```
or is there any reason why it couldn't work?